youtube_dl/extractor/vvvvid.py

```diff
@@ -45,16 +45,23 @@ def _real_initialize(self):
             'https://www.vvvvid.it/user/login',
             None, headers=self.geo_verification_headers())['data']['conn_id']
 
-    def _real_extract(self, url):
-        show_id, season_id, video_id = re.match(self._VALID_URL, url).groups()
+    def _download_info(self, url, video_id):
         response = self._download_json(
-            'https://www.vvvvid.it/vvvvid/ondemand/%s/season/%s' % (show_id, season_id),
-            video_id, headers=self.geo_verification_headers(), query={
+            url, video_id,
```

Comment:
remove the duplication as much as possible(`https://www.vvvvid.it/vvvvid/ondemand/%s/` is always used).